### PR TITLE
Add REST endpoints

### LINKED ISSUE #23: Implement REST Endpoints
- [ ] [POST `/game/create/`](https://github.com/SE750-DART/nerds-oppose-society/wiki/REST-Endpoints#post-gamecreate)
- [ ] [GET `/game/validate`](https://github.com/SE750-DART/nerds-oppose-society/wiki/REST-Endpoints#get-gamevalidate)
- [ ] [POST `/player/create`](https://github.com/SE750-DART/nerds-oppose-society/wiki/REST-Endpoints#post-playercreate)

See #19 for more

### PR DESCRIPTION
**Pull request by Alxjlv**
As per the title, the enpoints defined [here](https://github.com/SE750-DART/nerds-oppose-society/wiki/REST-Endpoints) have been implemented.

Resolves #23

### CODE REVIEW

File: backend/src/routes/player.route.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Router } from "express";
+import { createPlayer } from "../controllers/player.controller";
+
+const router = Router();
+
+router.get("/ping", async (req, res) => res.sendStatus(200));
```
Comment (taitfuller) [inline backend/src/routes/player.route.ts:6]: Were the ping routes just for testing?

File: backend/src/controllers/__tests__/player.controller.test.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+    expect(createSpy).toHaveBeenCalledTimes(1);
+
+    expect(status).toHaveBeenCalledTimes(1);
+    expect(status).toHaveBeenCalledWith(201);
+
+    expect(send).toHaveBeenCalledTimes(1);
+    expect(send).toHaveBeenCalledWith("69eb420cg69");
+  });
+  it("Responds with a 400 if the game code is invalid", async () => {
```
Comment (taitfuller) [inline backend/src/controllers/__tests__/player.controller.test.ts:51]: Hmmm looks like Prettier hasn't make the white spacing between tests consistent 🤔 super minor but could be nice to clean that up

File: backend/src/controllers/player.controller.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    if (typeof nickname != "string") {
+      res.status(400).send("Nickname not of type string");
+      return;
+    }
+    const dbResp = await createPlayerService(code, nickname);
+    res.status(201).send(dbResp);
+    return;
+  } catch (e) {
+    console.log(e);
```
Comment (taitfuller) [inline backend/src/controllers/player.controller.ts:25]: I think this log could be removed as we shouldn't need the log the `400` responses and the error handler will log `next(e)`

File: backend/src/controllers/player.controller.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    if (typeof code != "string") {
+      res.status(400).send("Code not of type string");
+      return;
+    }
+    if (typeof nickname != "string") {
+      res.status(400).send("Nickname not of type string");
+      return;
+    }
+    const dbResp = await createPlayerService(code, nickname);
+    res.status(201).send(dbResp);
```
Comment (taitfuller) [inline backend/src/controllers/player.controller.ts:22]: I think something like `playerId` would be a more descriptive name than `dbResp`

File: backend/src/controllers/game.controller.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  next: NextFunction
+): Promise<void> => {
+  try {
+    const code = req.query.gameCode;
+    if (typeof code != "string") {
+      res.status(400).send("Code not of type string");
+      return;
+    }
+    const dbResp = await validateGameCodeService(code);
+    if (dbResp) {
```
Comment (taitfuller) [inline backend/src/controllers/game.controller.ts:34]: I think something like `isValid` or `gameExists: boolean` would be more descriptive than `dbResp`

File: backend/src/controllers/game.controller.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+} from "../services/game.service";
+
+export const createGame = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+  try {
+    const dbResp = await createGameService();
+    console.log("Created game!");
```
Comment (taitfuller) [inline backend/src/controllers/game.controller.ts:14]: Should probably remove this log?

File: backend/src/controllers/game.controller.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    res.status(201).send(dbResp);
```
Comment (taitfuller) [inline backend/src/controllers/game.controller.ts:15]: I think `gameCode` would be a more descriptive name than `dbResp`

**Review by taitfuller — COMMENTED** [review]
Looks great man! Got a few minor comments but really nice work 👌🏼

File: backend/src/routes/player.route.ts (hunk 1 above)
Comment (Alxjlv) [reply backend/src/routes/player.route.ts:6]: Mostly yep - still useful but yeah mainly testing. Should I get rid of them?

File: backend/src/controllers/__tests__/player.controller.test.ts (hunk 2 above)
Comment (Alxjlv) [reply backend/src/controllers/__tests__/player.controller.test.ts:51]: Ooh shoot Prettier doing me dirty like that - I'll clean up some of them

File: backend/src/controllers/player.controller.ts (hunk 3 above)
Comment (Alxjlv) [reply backend/src/controllers/player.controller.ts:25]: Good catch

File: backend/src/controllers/player.controller.ts (hunk 4 above)
Comment (Alxjlv) [reply backend/src/controllers/player.controller.ts:22]: Yep fair

File: backend/src/routes/player.route.ts (hunk 1 above)
Comment (taitfuller) [reply backend/src/routes/player.route.ts]: Yeah I think we should remove them if they're not being used 👌🏼

**Review by taitfuller — APPROVED** [review]
Great work man! 🙌🏼 Looks awesome 😍

**Pull request merged**